fs/ext4/inode.c

```diff
@@ -3825,13 +3827,15 @@ static int __ext4_block_zero_page_range(handle_t *handle,
 
 	if (!buffer_uptodate(bh)) {
 		err = -EIO;
+		decrypt = S_ISREG(inode->i_mode) &&
```

Comment:
Based on the value of `decrypt` in `ext4_block_write_begin`, this should probably be the same:

```
			decrypt = S_ISREG(inode->i_mode) &&
				ext4_encrypted_inode(inode) &&
				!fscrypt_using_hardware_encryption(inode);
```

Reply:
It's different in the original commit though, I'm uncertain of changing it from that. There must've been some reason it's different, no? Especially since ext4 maintainer took a look at this.

Reply:
It's because of CAF's additions, c692cb96096def8727ae342ea0926c651f2eeed7 and 36c6512ecdc28e18befec67802af4ed075772112.

Reply:
> It's because of CAF's additions, [c692cb9](https://github.com/sonyxperiadev/kernel/commit/c692cb96096def8727ae342ea0926c651f2eeed7) and [36c6512](https://github.com/sonyxperiadev/kernel/commit/36c6512ecdc28e18befec67802af4ed075772112).

Ah, okai. The ICE commits. I see now. I'm going to do thsi as a seperate commit to the others as it diverges from the original.

Reply:
Done